cmd/dispatcher/main.go

```diff
@@ -110,6 +110,9 @@ func main() {
 			if errors.Is(err, context.Canceled) {
 				return
 			}
+			rmqHelper.CloseRabbitMQConnections(env.connection, logger)
+			time.Sleep(time.Second)
```

Comment:
I think the `Sleep`s are a bit inconsistent. We're doing it here as well as in the RabbitMQHelper which feels wrong. The way I see it, we want to sleep before doing a retry, regardless of the reason for the retry. So it might be better to contain that in the RabbitMQHelper specifically within the `SignalRetry`